src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.InfoCache.cs

```diff
@@ -69,6 +69,42 @@ public bool TryGetMemberInfo(IMethodSymbol method, out MemberInfo memberInfo)
                 return memberInfo.LengthLikeProperty != null;
             }
 
+            public bool TryGetMemberInfoOneArgument(IMethodSymbol method, out MemberInfo memberInfo)
+            {
+                if (!IsSliceLikeMethodWithOneArgument(method))
+                {
+                    memberInfo = default;
+                    return false;
+                }
+
+                if (_methodToMemberInfo.TryGetValue(method, out memberInfo))
+                {
+                    return memberInfo.LengthLikeProperty != null;
+                }
+
+                // Find overload of our method that is a slice-like method with two arguments.
+                // Computing member info for this method will also check that the containing type
+                // has an int32 'Length' or 'Count' property, and has a suitable indexer,
+                // so we don't have to.
+                var overloadWithTwoArguments = method.ContainingType
+                    .GetMembers(method.Name)
+                    .OfType<IMethodSymbol>()
+                    .FirstOrDefault(s => IsSliceLikeMethod(s));
```

Comment:
should we rename IsSliceLikeMethod to IsTwoArgumentSliceLikeMethod?

Reply:
Fixed; also renamed `IsSliceLikeMethodWithOneArgument` to `IsOneArgumentSliceLikeMethod`.